rust/chains/abacus-ethereum/src/multisig_ism.rs

```diff
@@ -0,0 +1,237 @@
+#![allow(clippy::enum_variant_names)]
+#![allow(missing_docs)]
+
+use std::collections::HashMap;
+use std::sync::Arc;
+use std::time::{Duration, SystemTime};
+
+use abacus_core::accumulator::merkle::Proof;
+use async_trait::async_trait;
+use ethers::abi::Token;
+use ethers::providers::Middleware;
+use ethers::types::{Selector, H160, H256, U256};
+use eyre::Result;
+use std::collections::hash_map::Entry::Occupied;
+use std::hash::Hash;
+use tokio::sync::Mutex;
+
+use abacus_core::{
+    AbacusAbi, AbacusContract, ChainCommunicationError, ContractLocator, MultisigIsm,
+    MultisigSignedCheckpoint,
+};
+
+use crate::contracts::multisig_ism::{MultisigIsm as EthereumMultisigIsmInternal, MULTISIGISM_ABI};
+use crate::trait_builder::MakeableWithProvider;
+
+#[derive(Debug)]
+struct Timestamped<Value> {
+    t: SystemTime,
```

Comment:
I think the more common practice for dealing with time elapsed between 2 events is to use [Instant](https://doc.rust-lang.org/std/time/struct.Instant.html), and you could then do `entry.get().t.elapsed() > self.expiry` down below in `ExpiringCache::get`

It also mostly solves the concern around the clock going backward: "Instants are always guaranteed, barring [platform bugs](https://doc.rust-lang.org/std/time/struct.Instant.html#monotonicity), to be no less than any previously measured instant when created"

Reply:
Ah yeah instant is indeed much better